### DISCUSSION TITLE
Run container with Podman

### DISCUSSION THREAD

**Discussion by dmothes** [photoprism/photoprism#697] [General]
Hi there, 

as askted in the forum. My way to get startet with podman. As i could start it as a user. 
As u have just installed podman as usual and done the intro to install it for use as a user you shoud start the container with 
https://wiki.archlinux.org/index.php/Podman#Rootless_Podman

1. You need e.g. to create your folders for photoprism

- config: /home/[user]/photoprism/.photoprism, 
`# mkdir -p ~/photoprism/.photoprism`
- than a orginals folder and maybe a import folder.
`# mkdir -p ~/photoprism/photoprism/original`
`# mkdir -p ~/photoprism/photoprism/import`
- next i load my library sortet to years/month-event, so u need this path too.
I did a nextcloud sync to a local folder and bind the subfolders seperatly to maybe be able to grow/shrink by insert or deletet some filepaths

- You need to clear [IP-to-bind] with your one or to clear this to connect to all local interfaces
- You need to replace the [user] in path or set your paths. My ones are an example

```
podman run \
-p [IP-to-bind]:2342:2342/tcp \
-e PHOTOPRISM_ADMIN_PASSWORD=please-change \
-e PHOTOPRISM_DEBUG=false \
-e PHOTOPRISM_PUBLIC=false \
-e PHOTOPRISM_READONLY=false \
-e PHOTOPRISM_UPLOAD_NSFW=true \
-e PHOTOPRISM_DETECT_NSFW=false \
-e PHOTOPRISM_EXPERIMENTAL=false \
-e PHOTOPRISM_SITE_URL=http://localhost:2342/ \
-e PHOTOPRISM_SITE_TITLE="My Photobook" \
-e PHOTOPRISM_SITE_CAPTION="Browse your life" \
-e PHOTOPRISM_SITE_DESCRIPTION \
-e PHOTOPRISM_SITE_AUTHOR \
-e PHOTOPRISM_HTTP_HOST=0.0.0.0 \
-e PHOTOPRISM_HTTP_PORT=2342 \
-e PHOTOPRISM_SETTINGS_HIDDEN=false \
-e PHOTOPRISM_DATABASE_DRIVER=sqlite \
-e PHOTOPRISM_SIDECAR_JSON=true \
-e PHOTOPRISM_SIDECAR_YAML=true \
-e PHOTOPRISM_THUMB_FILTER=lanczos \
-e PHOTOPRISM_THUMB_UNCACHED=false \
-e PHOTOPRISM_THUMB_SIZE=2048 \
-e PHOTOPRISM_THUMB_SIZE_UNCACHED=7680 \
-e PHOTOPRISM_JPEG_SIZE=7680 \
-e PHOTOPRISM_JPEG_QUALITY=92 \
-e PHOTOPRISM_DARKTABLE_PRESETS=false  \
--mount type=bind,source=/home/[user]/photoprism/.photoprism,destination=/photoprism/storage \
--mount type=bind,source=/mnt/[user]/photoprism/originals,destination=/photoprism/originals \
--mount type=bind,source=/mnt/[user]/images/2020,destination=/photoprism/originals/2020 \
--mount type=bind,source=/mnt/[user]/images/2019,destination=/photoprism/originals/2019 \
--mount type=bind,source=/mnt/[user]/images/2018,destination=/photoprism/originals/2018 \
--mount type=bind,source=/mnt/[user]/photoprism/import,destination=/photoprism/import \
photoprism/photoprism:latest
```

My next step will be to create a pod with a mysql database inside. 
One thing i have seen with my podman version 2.1.1. if i start it as a root user behind my firewall rules, it was missing to update the firewall for 'cstate new' rule

 #Regards

**Comment by graciousgrey** [comment]
Thank you for sharing how to setup PhotoPrism with Podman :)
We might add a section for different setup guides to our docs later.

**Comment by alexisLefebvre** [comment]
Should it be moved on the docs repo? https://github.com/photoprism/photoprism-docs

**Comment by graciousgrey** [comment]
I am not sure if people are looking at the discussions in the docs repo. But I would add it to the docs itself in a 
 new section "Setup Guides" :)

**Reply by alexisLefebvre** [reply]
My bad, I thought it was an issue on main repo. But it's a discussion and it fits here. :+1:
